### DISCUSSION TITLE
Calculating average with sqlalchemy.sql.functions

### DISCUSSION THREAD

**Discussion by GJena** [sqlalchemy/sqlalchemy#8254] [Usage Questions] [answered]
I'm using SQLAlchemy 1.4.37. I've seen examples of code that use 'avg' from functions to calculate the average. However, when I try to use it I get the error: AttributeError: module 'sqlalchemy.sql.functions' has no attribute 'avg'. I'm able to use sum and count successfully. Is avg not supported?

**Comment by CaselIT** [comment, marked as answer]
Hi,

You need to use the `func` attribute, not the module functions. You can use it as `sa.func.avg` or any other function

**Reply by GJena** [reply]
Thank you!
